server/src/room.ts

```diff
@@ -313,12 +359,41 @@ const verifyLockedOwner = async (
 ): Promise<boolean> => {
   const address = req.user ? req.user.payload.publicAddress.toLowerCase() : "";
 
-  const lockedOwnerAddress = await getLockedOwnerAddress(roomId);
-
-  if (lockedOwnerAddress && lockedOwnerAddress !== address.toLowerCase()) {
-    error(res, "unauthorized user for locked room");
+  const doc = await collection.doc(roomId).get();
+  if (!doc.exists) {
     return false;
   }
+  const docData = doc.data() as IChatRoom;
 
+  if(docData.isLocked){
+    if(!address){
+      error(res, "unauthorized user for locked room");
+      return false;
+    }
+
+    await axios.get('https://api.covalenthq.com/v1/137/address/' + address + '/balances_v2/?nft=true&key=ckey_c35e2c388e1b4efea8490fb8c83').then( async (result) => {
+     let permission = false;
+     for(let i = 0; i < result.data.data.items.length; i++){
+      if(NFT_CONTRACT_ADDRESS!.toLowerCase() === result.data.data.items[i].contract_address.toLowerCase()){
+         if(result.data.data.items[i].nft_data){
+           for(let j = 0; j < result.data.data.items[i].nft_data.length; j++){
+             const doc = await nftRooms.doc(result.data.data.items[i].nft_data[j].token_id).get();
+             const name = await doc.get("name");
+             if(name === roomId){
+               permission = true;
+             }
+           }
+         }
+       }
+     }
```

Comment:
I'd prefer if this were a separate method with a clear name explaining what it's doing. Currently it's very hard to read and understand what's happening, especially with nested for loops and very nested objects like `result.data.data.items[i].nft_data`

I'm not actually sure what this function is doing, can you explain it?